### ISSUE TITLE
Some Issues i had while trying this on one of my notebooks

### ISSUE THREAD

**Issue by omegabr99** [seeseemelk/gnome-1.0#1]
Sorry if this isnt enough but i dont have time rn

1 - the about buttom causes an crash
it gives an bad window error
2- tiles a broken
trying to apply one causes an crash
3 - menu editor causes an crash aswell

**Comment by seeseemelk** [comment]
Which programs are you trying to run exactly?

**Comment by omegabr99** [comment]
> Which programs are you trying to run exactly?

the panel with windowmaker e16 and fvwm2

**Comment by seeseemelk** [comment]
I haven't been able to properly replicate the crashes yet, but I've noticed that in panel many of the graphics are broken. This is caused by a bug in imlib 1.9.4. However, Arch Linux does ship imlib 1.9.15, but with `gdk_imlib.h` disabled which means gnome core won't build.
A temporary workaround is to download Arch's PKGBUILD for imlib and modifying it so that `gdk_imlib.h` is not disabled.

```
$ yay -G imlib
$ cd imlib
$ vim PKGBUILD
```

Within the `PKGBUILD` file you'll want to remove the following line:
```
patch -p1 -i "${srcdir}/imlib-gtk.patch"
```

Then build and install
```
$ makepkg -i
```

Can you see if it still crashes with a patched imlib?

**Comment by omegabr99** [comment]
> I haven't been able to properly replicate the crashes yet, but I've noticed that in panel many of the graphics are broken. This is caused by a bug in imlib 1.9.4. However, Arch Linux does ship imlib 1.9.15, but with `gdk_imlib.h` disabled which means gnome core won't build.
> A temporary workaround is to download Arch's PKGBUILD for imlib and modifying it so that `gdk_imlib.h` is not disabled.
> 
> ```
> $ yay -G imlib
> $ cd imlib
> $ vim PKGBUILD
> ```
> 
> Within the `PKGBUILD` file you'll want to remove the following line:
> 
> ```
> patch -p1 -i "${srcdir}/imlib-gtk.patch"
> ```
> 
> Then build and install
> 
> ```
> $ makepkg -i
> ```
> 
> Can you see if it still crashes with a patched imlib?

sorry for taking long to awnser but i will try

**Comment by omegabr99** [comment]
`misc.c: In function ‘gdk_imlib_set_color_map’:
misc.c:69:3: error: unknown type name ‘GdkWindowPrivate’; did you mean ‘GdkWindowState’?
   69 |   GdkWindowPrivate   *private;
      |   ^~~~~~~~~~~~~~~~
      |   GdkWindowState
misc.c:79:14: error: ‘GdkWindowPrivate’ undeclared (first use in this function); did you mean ‘GdkWindowState’?
   79 |   private = (GdkWindowPrivate *) & gdk_root_parent;
      |              ^~~~~~~~~~~~~~~~
      |              GdkWindowState
misc.c:79:14: note: each undeclared identifier is reported only once for each function it appears in
misc.c:79:32: error: expected expression before ‘)’ token
   79 |   private = (GdkWindowPrivate *) & gdk_root_parent;
      |                                ^
misc.c:84:47: error: request for member ‘xwindow’ in something not a structure or union
   84 |       cm = XCreateColormap(id->x.disp, private->xwindow,
      |                                               ^~
misc.c:94:56: error: request for member ‘xwindow’ in something not a structure or union
   94 |   id->x.base_window = XCreateWindow(id->x.disp, private->xwindow,
      |                                                        ^~
make[2]: *** [Makefile:698: misc.lo] Erro 1
make[2]: Saindo do diretório '/home/omegabr/imlib/src/imlib-1.9.15/gdk_imlib'

make[1]: *** [Makefile:591: all-recursive] Erro 1
make[1]: Saindo do diretório '/home/omegabr/imlib/src/imlib-1.9.15'

make: *** [Makefile:395: all] Erro 2`

**Comment by seeseemelk** [comment]
Can you run it with GDB to see where it crashes and post the stack trace here? I've noticed that `gedit` crashes because it tries to load a font that doesn't exist, resulting in a null dereference. `panel` seems to be working fine for me, so it might be a similar issue.

**Comment by seeseemelk** [comment]
Aight, I've been able to play around with it and it does indeed crash quite often. In fact, it always seems to crash when I select a second menu item.
I've been able to run it through gdb, and it seems to crash at some `gnome_dentry_set...` function, but I haven't yet been able to fix it.

Thanks for reporting this though! It's fun to know that I'm not the only one who's interested in running ancient open source 90s-looking software!

**Comment by omegabr99** [comment]
> Aight, I've been able to play around with it and it does indeed crash quite often. In fact, it always seems to crash when I select a second menu item.
> I've been able to run it through gdb, and it seems to crash at some `gnome_dentry_set...` function, but I haven't yet been able to fix it.
> 
> Thanks for reporting this though! It's fun to know that I'm not the only one who's interested in running ancient open source 90s-looking software!

no problem i cant wait to use this again

**Comment by seeseemelk** [comment]
I've attempted to create a fix in the `fix/gmenu-crash` branch (bddf5a57e76325c79de9a27d5349c517341e28b5), but I don't think it really solves the underlying issue. In fact, if I select an icon for a program I still get a crash, and I get warnings of incorrect reference counts when running from a terminal.
This will need more investigating, but it's not a pretty - or modern - codebase to try and fix. And to top it off, it's near impossible to find any documentation for it given that it's 22 year old software.

**Comment by seeseemelk** [comment]
This should be now be properly fixed by 7aaa26c92cb9d8efe6a105902969d3a55f1dd618. I can no longer get it to crash.
If it isn't fixed for you, feel free to reopen this issue.
